sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowFunctionFrame.scala

```diff
@@ -200,15 +200,19 @@ class FrameLessOffsetWindowFunctionFrame(
 
   override def prepare(rows: ExternalAppendOnlyUnsafeRowArray): Unit = {
     resetStates(rows)
-    if (ignoreNulls) {
-      findNextRowWithNonNullInput()
+    if (Math.abs(offset) > rows.length) {
+      inputIndex = offset
     } else {
-      // drain the first few rows if offset is larger than zero
-      while (inputIndex < offset) {
-        if (inputIterator.hasNext) inputIterator.next()
-        inputIndex += 1
+      if (ignoreNulls) {
```

Comment:
seems we can just change this to `if (ignoreNulls && Math.abs(offset) > rows.length)`?

Reply:
when `offset > rows.length` it is safe to just set `inputIndex` to its value. I can add conditional inside `if (ignoreNulls)` if you feel strongly about it :)